src/Symfony/Component/Form/Guess/Guess.php

```diff
@@ -62,7 +62,7 @@ abstract class Guess
      *
      * @param Guess[] $guesses An array of guesses
      *
-     * @return self|null
+     * @return static|null
```

Comment:
Current class (on which we can call static method) and returned not related. Isn't it?

AFAIK return type relates on provided `$guesses` items

Reply:
I forgot to change also the `$guesses` items, here an example without the change:

https://phpstan.org/r/3b1299cc-0c60-471b-a70d-a776f8a01645

Reply:
Guess class can be used itself for selection some different implementations. Also, we can use any inherited class for same - not only between selecting same classes as called - https://phpstan.org/r/d7e6e6e8-463e-4c3a-a576-6c9dbe87d81c

Reply:
To solve your example templates can be used - https://phpstan.org/r/7eba3ff1-a47f-45da-a0bb-99fae63c3597